src/App.js

```diff
@@ -1,35 +1,51 @@
 import React from 'react';
+import AddTodoForm from './AddTodoForm';
+import TodoList from './TodoList';
 
 
-let todoList =[
-  {
-   id: 1,
-    title: "Call your Mother Again",
-   },
-  {
-   id: 2,
-    title: "Water the plants",
-   },
-  {
-     id: 3,
-     title: "Find a Loft",
-   },
-   ];
-
-   function App() {
-    return (
-      <div style={{ textAlign: 'center' }}>
-        <h1>Todo List</h1>
-        <ul>
-          {todoList.map(item =>
-            <li key={item.id}>
-              <li>{item.title}</li>
-            </li>
-            )}
-  
-        </ul>
-      </div>
-    );
-  }
-  
-  export default App;
\ No newline at end of file
+//Not Easy 
+// Above the App functional component, create a new function named useSemiPersistentState which will be a custom hook
+
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = React.useState(
+
+    //Update the default state for todoList to read your "savedTodoList" item from localStorage
+
+    //  o	Hint: JSON.parse method  Update your default state to parse the value of the "savedTodoList" item
+    JSON.parse(localStorage.getItem('savedTodoList'))
+  );
+
+//   •	  Open /src/App.js
+// •	  Define a useEffect React hook with todoList as a dependency
+// •	  Inside the side-effect handler function, save the todoList inside localStorage with the key "savedTodoList"
+
+  React.useEffect(() => {
+
+//     Update your side-effect function to convert todoList to a string before saving in localStorage
+// o	Hint: JSON.stringify method
+
+    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+  }, [todoList]);
+
+  return [todoList, setTodoList];
+};
+
+function App() {
+  const [todoList, setTodoList] = useSemiPersistentState();
+
+
+  // Add a return statement in useSemiPersistentState that returns the todoList state variable and setter in an Array 
+  const addTodo = (newTodo) => {
+    setTodoList([...(todoList || []), newTodo]);
+  };
+
+  return (
+    <div>
```

Comment:
The last part of the assignment wanted you to use `<React.Fragment>` -- React Fragments help you avoid certain errors and prevents "div soup" (Basically having a ton of necessary divs in your final code).

https://reactjs.org/docs/fragments.html